geniza/annotations/signals.py

```diff
@@ -0,0 +1,44 @@
+from django.conf import settings
+from django.contrib.admin.models import ADDITION, LogEntry
+from django.contrib.auth.models import User
+from django.contrib.contenttypes.models import ContentType
+
+from geniza.corpus.models import Document
+from geniza.footnotes.models import Footnote, Source
+
+
+def update_footnote(instance, **kwargs):
+    """On Annotation save, update related footnote's doc_relation field if necessary"""
+    # get pgpid from annotation content and match it to a Document
+    document = Document.from_manifest_uri(
+        instance.content["target"]["source"]["partOf"]["id"]
+    )
+
+    # create or update footnote
+    try:
+        # if the footnote exists but is an EDITION, make it also a DIGITAL_EDITION
+        footnote = document.footnotes.exclude(
+            doc_relation__contains=Footnote.DIGITAL_EDITION
+        ).get(
+            source=Source.from_uri(instance.content["dc:source"]),
+            doc_relation__contains=Footnote.EDITION,
+        )
+        footnote.doc_relation.append(Footnote.DIGITAL_EDITION)
+        footnote.save()
+    except Footnote.DoesNotExist:
+        # create the DIGITAL_EDITION footnote (unless it already exists)
+        footnote, created = document.footnotes.get_or_create(
+            source=Source.from_uri(instance.content["dc:source"]),
+            doc_relation__contains=Footnote.DIGITAL_EDITION,
+        )
+        if created:
+            footnote.doc_relation = [Footnote.DIGITAL_EDITION]
+            footnote.save()
+            LogEntry.objects.log_action(
+                # can't get user in a signal handler, so associate with script
+                user_id=User.objects.get(username=settings.SCRIPT_USERNAME).id,
+                content_type_id=ContentType.objects.get_for_model(Footnote).pk,
+                object_id=footnote.pk,
+                object_repr=str(footnote),
+                action_flag=ADDITION,
+            )
```

Comment:
Sorry I wasn't clearer about the logic I want! I need to write down the big picture somewhere asap, as much as I have it figured out; maybe when you review you can help me identify any gaps I need to still fill in.

The digital edition footnotes should be separate from the edition footnotes. For the sake of this signal handler, we can assume that the appropriate edition footnotes have already been converted to digital editions by the tei to annotation script. 

For this signal handler, I think we just need to make sure that a digital edition footnote for the given source + document exists.  We can't use `get_or_create` because that will create a duplicate footnote if any of the fields differ (e.g., a note documenting emendations by a second contributor).

Did you do away with the delete signal handler? I think we still need that.

Reply:
> Did you do away with the delete signal handler? I think we still need that.

I was asking about that [here](https://github.com/Princeton-CDH/geniza/pull/1006#discussion_r942861472), what should we do on delete if it's the only annotation?